src/SLIP10Node.test.ts

```diff
@@ -0,0 +1,344 @@
+      const node = await SLIP10Node.create({
+        depth: 3,
+        key: Buffer.alloc(64).fill(2).toString('base64'),
+        curve: secp256k1,
+      });
+
+      expect(node.key).toHaveLength(88);
+      expect(node.depth).toStrictEqual(3);
+      expect(node.toJSON()).toStrictEqual({
+        depth: 3,
+        key: node.key,
+      });
+    });
+
+    it('initializes a new node (depth, hex string key)', async () => {
+      const node = await SLIP10Node.create({
+        depth: 3,
+        key: Buffer.alloc(64).fill(2).toString('hex'),
+        curve: secp256k1,
+      });
+
+      expect(node.key).toHaveLength(88);
+      expect(node.depth).toStrictEqual(3);
+      expect(node.toJSON()).toStrictEqual({
+        depth: 3,
+        key: node.key,
+      });
+    });
+
+    it('initializes a new node (depth, 0x-prefixed hex string key)', async () => {
+      const node = await SLIP10Node.create({
+        depth: 3,
+        key: `0x${Buffer.alloc(64).fill(2).toString('hex')}`,
+        curve: secp256k1,
+      });
+
+      expect(node.key).toHaveLength(88);
+      expect(node.depth).toStrictEqual(3);
+      expect(node.toJSON()).toStrictEqual({
+        depth: 3,
+        key: node.key,
+      });
+    });
+
+    it('throws if the depth is invalid', async () => {
+      const validBufferKey = Buffer.alloc(64).fill(1);
+      const inputs = [
+        -1,
+        0.1,
+        -0.1,
+        NaN,
+        Infinity,
+        '0',
+        'zero',
+        {},
+        null,
+        undefined,
+      ];
+
+      for (const input of inputs) {
+        await expect(() =>
+          SLIP10Node.create({
+            depth: input as any,
+            key: validBufferKey,
+            curve: secp256k1,
+          }),
+        ).rejects.toThrow(
+          `Invalid HD tree path depth: The depth must be a positive integer. Received: "${input}"`,
+        );
+      }
+    });
+
+    it('throws if both a derivation path and a depth are specified', async () => {
+      await expect(() =>
+        SLIP10Node.create({
+          depth: 2, // This is the correct depth, but it's still forbidden
+          derivationPath: [
+            defaultBip39NodeToken,
+            BIP44PurposeNodeToken,
+            `bip32:60'`,
+          ],
+          curve: secp256k1,
+        }),
+      ).rejects.toThrow(
+        'Invalid parameters: May not specify a depth if a derivation path is specified. The depth will be calculated from the path.',
+      );
+    });
+
+    it('throws if neither a derivation path nor a key is specified', async () => {
+      await expect(() =>
+        SLIP10Node.create({ depth: 1, curve: secp256k1 }),
+      ).rejects.toThrow(
+        'Invalid parameters: Must specify either key or derivation path.',
+      );
+    });
+
+    it('throws if the derivation path is empty', async () => {
+      await expect(() =>
+        SLIP10Node.create({ derivationPath: [] as any, curve: secp256k1 }),
```

Comment:
You could constraint the type using [non-zero arrays](https://stackoverflow.com/a/56006703)

Reply:
It is constrained, hence the type cast here. Without it it would result in TypeScript errors.